debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/DumpModuleCommand.java

```diff
@@ -0,0 +1,226 @@
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper.ClassOutput;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper.ModuleIteratorFilter;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper.ModuleOutput;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper.PackageIteratorFilter;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper.PackageOutput;
+
+import static com.ibm.j9ddr.vm29.tools.ddrinteractive.commands.DumpModuleCommand.Status.*;
+
+public class DumpModuleCommand extends Command 
+{
+	enum Status {
+		CLASS, MODULE, PACKAGE, ALL, HELP
+	}
+
+	public DumpModuleCommand() 
+	{
+		addCommand("dumpmodule", "[all|requires|exports|classes|packages] <moduleAddress>|help", "List details about a provided module");
+	}
+	
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException 
+	{
+		Status status = HELP;
+		ClassIteratorFilter classFilter = null;
+		ClassOutput classOutput = null;
+		ModuleIteratorFilter moduleFilter = null;
+		ModuleOutput moduleOutputter = null;
+		PackageIteratorFilter packageFilter = null;
+		PackageOutput packageOutputter = null;
+		String filterArg = null;
+		switch (args.length) {
+		case 1:
+			if (!args[0].equalsIgnoreCase("help")) {
+				filterArg = args[0];
+				status = ALL;
+			} /* else stay status.HELP */
+			break;
+		case 2:
+			filterArg = args[1];
+			if (args[0].equalsIgnoreCase("all")) {
+				status = ALL;
+			} else if (args[0].equalsIgnoreCase("requires")) {
+				moduleFilter = DumpModuleCommand::filterModuleRequired;
+				moduleOutputter = ModularityHelper::printJ9Module;
+				status = MODULE;
+			} else if (args[0].equalsIgnoreCase("exports")) {
+				packageFilter = DumpModuleCommand::filterPackageModuleAndExport;
+				packageOutputter = ModularityHelper::printPackageExportVerbose;
+				status = PACKAGE;
+			} else if (args[0].equalsIgnoreCase("classes")) {
+				classFilter = DumpModuleCommand::filterClassByModule;
+				classOutput = ModularityHelper::printJ9Class;
+				status = CLASS;
+			} else if (args[0].equalsIgnoreCase("packages")) {
+				packageFilter = DumpModuleCommand::filterPackageModule;
+				packageOutputter = ModularityHelper::printJ9Package;
+				status = PACKAGE;
+			} /* else stay status.HELP */
+			break;
+		default:
+			break; /* default stay status.HELP */
+		}
+		try {
+			J9ModulePointer modulePtr = J9ModulePointer.cast(Long.decode(filterArg));
+			switch (status) {
+			case CLASS:
+				J9ClassLoaderPointer classLoaderPtr = modulePtr.classLoader();
+				ModularityHelper.iterateClassLoaderClasses(out, classFilter, classOutput, filterArg, classLoaderPtr);
+				break;
+			case MODULE:
+				ModularityHelper.iterateModules(out, moduleFilter, moduleOutputter, filterArg);
+				break;
+			case PACKAGE:
+				ModularityHelper.iteratePackages(out, packageFilter, packageOutputter, filterArg);
```

Comment:
you can also iterate only the classloader packages here

Reply:
this comment should really be in the dumpmodule PR

Reply:
Moved to dumpmodule PR